Trigger mimir workflow on branch pushes

dimitarvdimitrov opened the pull request:
This workflow automatically merges changes from upstream prometheus/prometheus branches into local branches.

Features:
- Manual dispatch with configurable upstream/local branches (default: main/main)
- Daily scheduled runs at 02:00 UTC
- Proper conflict detection with placeholders for Slack notifications
- Creates PRs only when merge is successful
- Skips when already up-to-date

The workflow is based on the existing cherry-pick workflow but handles full branch merges instead of individual commits.

alexweav commented on .github/workflows/merge-upstream-prometheus.yml:
```diff
@@ -0,0 +1,290 @@
+      - name: Add prometheus/prometheus remote and fetch
+        run: |
+          git remote add prometheus https://github.com/prometheus/prometheus.git
+          git fetch prometheus ${{ needs.check-merge.outputs.upstream_branch }}
```
Huge nit: Should we fetch and merge prometheus at the `$UPSTREAM_COMMIT` stored in `$GITHUB_OUTPUT`?
In the unlikely event that a new prometheus commit gets merged between `check-merge` and `attempt-merge`, repeating a fetch might not put us where we think we are.

TBF I think the worst that can happen is the PR has the wrong commit mentioned in it, so this is low prio.

alexweav commented on .github/workflows/merge-upstream-prometheus.yml:
```diff
@@ -0,0 +1,290 @@
+          channel-id: C04AF91LPFX #mimir-ci-notifications
+          payload: |
+            {
+              "text": ":warning: *Merge conflicts detected in upstream prometheus merge*\n\n*Details:*\n• Upstream branch: `${{ needs.check-merge.outputs.upstream_branch }}`\n• Local branch: `${{ needs.check-merge.outputs.local_branch }}`\n• Latest upstream commit: <https://github.com/prometheus/prometheus/commit/${{ needs.check-merge.outputs.upstream_commit }}|${{ needs.check-merge.outputs.upstr ...
```
Nit: Maybe worth pinging `@mimir-devops-rollout`?

dimitarvdimitrov commented:
@alexweav Thanks for the excellent feedback\! I've implemented both suggestions and added a significant improvement:

## Changes Made

### 1. ✅ Use upstream commit hash instead of branch references
- Now uses `${{ needs.check-merge.outputs.upstream_commit }}` for all merge operations
- Avoids repeating fetch logic and ensures we merge the exact commit that was checked

### 2. ✅ Added @mimir-devops-rollout team mentions
- Slack notifications now ping `<\!subteam^S03C2P8659U>` for both conflicts and errors
- Teams get notified immediately when manual intervention is needed

### 3. 🚀 **New**: Draft PR creation for conflicts (major UX improvement)
Instead of just failing when conflicts occur, the workflow now:
- **Creates a commit with conflict markers preserved** (the actual `<<<<<<<`, `=======`, `>>>>>>>` in files)
- **Opens a draft PR** with detailed resolution instructions
- **Includes step-by-step conflict resolution guide** in PR description
- **Sends Slack notification with direct PR link** for immediate action
- **Provides much better starting point** for manual conflict resolution

This addresses the common pain point where conflicts would just fail the workflow, leaving developers to manually recreate the merge attempt. Now they get a ready-to-edit branch with clear guidance.

The workflow is much more developer-friendly and should significantly reduce the friction when upstream conflicts occur. What do you think of this approach?
